### PR TITLE
prov/efa: Add tracepoints to send/recv/read ops

### PR DESCRIPTION
**Pull request by jtamzn**
This PR adds 9 tracepoints into EFA provider. Detailed tracepoints are listed in its docs `prov/efa/docs/efa_tracing.txt`

Signed-off-by: Tang, Jingyin <jytang@amazon.com>

### CODE REVIEW

File: prov/efa/src/rxr/efa_tp_def.h [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+
+#define X_ENTRY_ARGS \
+	int, msg_id, \
+	size_t, ctx, \
+	int, total_len
+
+#define X_ENTRY_FIELDS \
+	lttng_ust_field_integer(int, msg_id, msg_id) \
+	lttng_ust_field_integer(size_t, ctx, ctx) \
+	lttng_ust_field_integer(int, totel_len, total_len)
```
Comment (ryanamazon) [inline prov/efa/src/rxr/efa_tp_def.h:20]: total_len

File: prov/efa/src/rxr/efa_tp_def.h [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+	read_completed,
+	LTTNG_UST_TP_ARGS(
+		char*, entry_name,
+		X_ENTRY_ARGS,
+		size_t, context
+	),
+	LTTNG_UST_TP_FIELDS(
+		lttng_ust_field_string(entry_name, entry_name)
+		X_ENTRY_FIELDS
+		lttng_ust_field_integer(size_t, context, context)
```
[the remark this answers is not in the document]
Comment (jtamzn) [reply prov/efa/src/rxr/efa_tp_def.h:187]: Fixed

File: prov/efa/src/rxr/efa_tp_def.h [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+		X_ENTRY_FIELDS
+	)
+)
+LTTNG_UST_TRACEPOINT_LOGLEVEL(fi_efa_prov, runtread_read_posted, LTTNG_UST_TRACEPOINT_LOGLEVEL_INFO)
+
+LTTNG_UST_TRACEPOINT_EVENT(
+	fi_efa_prov,
+	read_completed,
+	LTTNG_UST_TP_ARGS(
+		char*, entry_name,
```
[the remark this answers is not in the document]
Comment (jtamzn) [reply prov/efa/src/rxr/efa_tp_def.h:180]: It is unnecessary anymore, for simplicity, this field is removed.

File: prov/efa/src/rxr/efa_tp_def.h (hunk 1 above)
Comment (jtamzn) [reply prov/efa/src/rxr/efa_tp_def.h:20]: Fixed

File: prov/efa/src/rxr/efa_tp_def.h [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+	lttng_ust_field_integer(size_t, ctx, ctx) \
+	lttng_ust_field_integer(int, totel_len, total_len) 
+
+#define MSG_ARGS \
+	size_t, msg_ctx, \
+	size_t, addr
+
+#define MSG_FIELDS \
+	lttng_ust_field_integer(size_t, msg_ctx, msg_ctx) \
+	lttng_ust_field_integer(size_t, addr, addr)
```
[the remark this answers is not in the document]
Comment (jtamzn) [reply prov/efa/src/rxr/efa_tp_def.h]: fixed

**Comment by wzamazon** [comment]
bot:aws:retest

**Review by wzamazon — APPROVED** [review]

**Comment by wzamazon** [comment]
bot:aws:retest

**Comment by wzamazon** [comment]
bot:aws:retest

File: prov/efa/src/rxr/rxr_msg.c [hunk 5]
```diff
@@ -875,6 +884,7 @@ int rxr_msg_proc_unexp_msg_list(struct rxr_ep *ep, const struct fi_msg *msg,
 				uint64_t tag, uint64_t ignore, uint32_t op, uint64_t flags,
 				struct rxr_rx_entry *posted_entry)
 {
+	perror("------ Called!\n");
```
Comment (wzamazon) [inline prov/efa/src/rxr/rxr_msg.c]: Remove this line and force push, instead of add 2nd commit to fix it.

**Review by wzamazon — APPROVED** [review]

**Comment by ryanamazon** [comment]
> @ryanamazon Did all your comments get addressed?

Yes, with one possible remark.  Thank you!

**Pull request merged**